client/my-sites/checkout/composite-checkout/lib/normalize-transaction-response.ts

```diff
@@ -1,24 +1,26 @@
 import { WPCOMTransactionEndpointResponse } from '@automattic/wpcom-checkout';
 
-const emptyResponse: WPCOMTransactionEndpointResponse = {
+export interface FailedResponse {
+	success: false;
+}
+
+const emptyResponse: FailedResponse = {
 	success: false,
-	error_code: '',
-	error_message: '',
 };
 
 export default function normalizeTransactionResponse(
 	response: unknown
-): WPCOMTransactionEndpointResponse {
+): WPCOMTransactionEndpointResponse | FailedResponse {
 	if ( ! response ) {
 		return emptyResponse;
 	}
 	const transactionResponse = response as WPCOMTransactionEndpointResponse;
 	if (
-		transactionResponse.message ||
+		( 'message' in transactionResponse && transactionResponse.message ) ||
```

Comment:
Great approach here! From my understanding, this can limit runtime errors from accessing the message property if it does not exist.

Reply:
Yup! And more importantly it "convinces" TypeScript that `transactionResponse.message` is ok to access (although we have to do both because `'message' in transactionResponse` will return true even if `transactionResponse.message` is undefined).